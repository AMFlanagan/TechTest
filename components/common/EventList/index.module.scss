.list {
  width: 100%;
  max-width: 800px;
  margin: 0 auto;
}

.title {
  padding: 8px 0;
  color: green;
}

.eventsContainer {
  max-width: 800px;
  margin: 0 auto;
}
