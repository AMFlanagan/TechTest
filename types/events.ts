export enum Sport {
  FOOTBALL = 'football',
  RUGBY = 'rugby',
  HORSE_RACING = 'horse-racing',
}

export enum FootballMarket {
  RESULT = 'Result',
  NEXT_GOAL = 'Next goal',
}

export enum RugbyMarket {
  RESULT = 'Result',
  NEXT_TRY = 'Next Try',
}

export enum HorseRacingMarket {
  RESULT = 'Result',
}

export interface IOutcome {
  id: number
  name: string
  odds: number
}

export type Market = FootballMarket | HorseRacingMarket | RugbyMarket

export interface IMarket {
  type: Market
  outcomes: IOutcome[]
}

export interface IEvent {
  id: number
  sport: Sport
  eventName: string
  markets: IMarket[]
  startTime: string
}
