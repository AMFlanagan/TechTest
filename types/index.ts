export interface IEvent {
  marketId: number
  outcomes: string[]
}
