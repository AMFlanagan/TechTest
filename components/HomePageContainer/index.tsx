import React from 'react'
import { IEvent, Sport } from '../../types'

import EventList from '../common/EventList'

interface IProps {
  allEvents: { [key in Sport]: IEvent[] }
}

const HomePage: React.FC<IProps> = (props: IProps) => {
  return (
    <div>
      {Object.entries(props.allEvents).map((sport, i: number) => {
        const [title, events] = sport
        return <EventList key={i} title={title} events={events} />
      })}
    </div>
  )
}

HomePage.displayName = 'HomePage'

export default HomePage
