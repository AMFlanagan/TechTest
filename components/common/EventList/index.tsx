import React from 'react'
import { IEvent } from '../../../types'

import Event from '../Event'

import styles from './index.module.scss'

interface IProps {
  events: IEvent[]
}

const EventList: React.FC<IProps> = (props: IProps) => {
  return (
    <div className={styles.list}>
      <div className={styles.eventsContainer}>
        {props.events.map((event: IEvent, i: number) => {
          return <Event key={i} event={event} />
        })}
      </div>
    </div>
  )
}

EventList.displayName = 'EventList'

export default EventList
