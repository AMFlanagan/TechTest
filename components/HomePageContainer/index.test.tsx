import HomePageContainer from '.'
import { render } from '@testing-library/react'
import { IEvent, Sport } from '../../types'
import EventList from '../common/EventList'

jest.mock('../common/EventList', () => jest.fn(() => <div>EventList</div>))

describe('HomePageContainer', () => {
  it('renders as expect', () => {
    const allEvents: IEvent[] = [
      {
        marketId: 1,
        outcomes: ['Team A Win', 'Draw', 'Team B Win'],
      },
      {
        marketId: 2,
        outcomes: ['Team C Win', 'Draw', 'Team D Win'],
      },
      {
        marketId: 3,
        outcomes: ['Team E Win', 'Draw', 'Team F Win'],
      },
    ]

    const { asFragment } = render(<HomePageContainer allEvents={allEvents} />)

    expect(EventList).toBeCalledWith(
      expect.objectContaining({
        events: allEvents,
      }),
      expect.anything()
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
