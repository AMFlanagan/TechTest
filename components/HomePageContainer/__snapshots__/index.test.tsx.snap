// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageContainer renders as expect 1`] = `
<DocumentFragment>
  <div>
    <div>
      EventList
    </div>
  </div>
</DocumentFragment>
`;
